test/test_cuda.py

```diff
@@ -1595,6 +1595,22 @@ def test_norm_type_conversion(self):
         a = torch.ones(65536).cuda().half()
         self.assertEqual(a.norm(p=0, dtype=torch.float32), 65536)
 
+    def test_cuda_randperm(self):
+        for rng in [torch.Generator(device="cuda:0"), torch.Generator(device="cuda")]:
+            # should not raise RuntimeError if device type match
+            try:
+                torch.randperm(n=50, generator=rng, device="cuda")
+            except RuntimeError as e:
+                self.fail(e)
```

Comment:
Why do we need `try except:` block here in the first place? Any uncaught exception automatically result in test failure, isn't it?
```suggestion
            torch.randperm(n=50, generator=rng, device="cuda")
```